python/feathub/processors/spark/dataframe_builder/spark_dataframe_builder.py

```diff
@@ -322,11 +340,48 @@ def _get_dataframe_from_derived_feature_view(
                     f"{type(feature.transform).__name__} for feature {feature.name}."
                 )
 
+        tmp_dataframe = self._apply_filter_if_any(
```

Comment:
Since this method is only meaningful when `filter_expr` is not None, it seems more reasonable to invoke it only when `filter_expr` is not None.

Also, since this method is only invoked in one place and it is simpler, maybe we should just inline it.